query.sql

```diff
@@ -94,22 +112,39 @@ SELECT
   COUNTIF(COALESCE(environment.services.account_enabled, FALSE)) > 0 AS sync_signed_in,
   COUNTIF(COALESCE(payload.processes.parent.scalars.formautofill_credit_cards_autofill_profiles_count IS NOT NULL, FALSE)) > 0 AS ccards_saved,
   COUNTIF(COALESCE(payload.processes.parent.scalars.dom_parentprocess_private_window_used, FALSE)) > 0 AS pbm_used,
-  COUNTIF(COALESCE(ARRAY_LENGTH(payload.keyed_histograms.FX_MIGRATION_HISTORY_QUANTITY), 0) > 0) > 0 AS imported_history,
-  COUNTIF(COALESCE(ARRAY_LENGTH(payload.keyed_histograms.FX_MIGRATION_BOOKMARKS_QUANTITY), 0) > 0) > 0 AS imported_bookmarks,
-  COUNTIF(COALESCE(ARRAY_LENGTH(payload.keyed_histograms.fx_migration_logins_quantity), 0) > 0 ) > 0 AS imported_logins,
-  SUM(COALESCE(payload.processes.parent.scalars.browser_engagement_max_concurrent_tab_pinned_count, 0)) AS pinned_tab_count_count,
-  SUM(COALESCE(ARRAY_LENGTH(payload.processes.parent.keyed_scalars.browser_ui_interaction_preferences_pane_general), 0)) AS unique_preferences_accessed_count,
-  SUM(COALESCE((SELECT SUM(value) FROM UNNEST(payload.processes.parent.keyed_scalars.browser_ui_interaction_preferences_pane_general)), 0)) AS preferences_accessed_total,
-  SUM(COALESCE(ARRAY_LENGTH(payload.processes.parent.keyed_scalars.browser_ui_interaction_bookmarks_bar), 0)) AS unique_bookmarks_bar_accessed_count,
-  SUM(COALESCE((SELECT SUM(value) FROM UNNEST(payload.processes.parent.keyed_scalars.browser_ui_interaction_bookmarks_bar)), 0)) AS bookmarks_bar_accessed_total,
+  
+  # [@shong] the imports query is actually wrong - the histograms include imports from profile refreshes (aka profile resets) 
+  # [@shong] which is where a lot of these import events are coming from. These are indicated by source: Firefox, but it needs 
+  # [@shong] to be filtered out. See: https://colab.research.google.com/drive/1yauD-2JcfvvFt_P87JfzSQ0tCEWHTZ4w
+  # [@shong] also, I think the histogram might have bins even if there are no items imported, so we should be checking
+  # [@shong] if the values are greater then 0, not array length. 
+  # [@shong] I'll add the correct query to this later (not sure if I can do it in this iteration, but put that action item on me) 
+  # [@shong] removed: imported_history, imported_bookmarks, imported_logins
+
+  # [@shong] so the pinned_tab_count_count had a bug with recording (it was only recording when a new pin / unpin action happened, 
+  # [@shong] not recording state), see: https://bugzilla.mozilla.org/show_bug.cgi?id=1639292 
+  # [@shong] so it looks like it got resolved, but it's not clear to me from the comments if the telemetry is doing exactly what we
+  # [@shong] think it does, or something similar. We should follow up to confirm exact behavior currently before adding. 
+  # [@shong] I've ping'd andrei for clarification. 
+  # [@shong] removed: pinned_tab_count_count
```

Comment:
update, i checked in with Andrei who worked on the linked bug. 

So looks like the bug in this telemetry was resolved, the behavior is as follows: 
* It will initially report the number of pinned tabs from the restore state at startup
* It will increment if new tabs are pinned
* It will NOT decrement if tabs are unpinned

So it's slightly wrong.. but probably okay. This metric can probably be included now, I'm thinking